packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -490,10 +504,20 @@ export default createRule<Options, MessageId>({
         );
 
         if (isVoidReturningFunctionType(checker, tsNode.name, contextualType)) {
-          context.report({
-            node: node.value,
-            messageId: 'voidReturnProperty',
-          });
+          const functionNode = findFunctionNode(node.value);
```

Comment:
It doesn't seem to me that the `findFunctionNode` function should be necessary here... We _know_ that the ESTree `node` is a property that, according to the TS AST, is a method declaration. So, we shouldn't need to recurse to the enclosing function node; we should be infer directly where the function node would be. Seemingly, 

```ts
const x = {
    f() { } // <-- functionNode _is_ `node.value`
}
```
not sure if there other cases to be aware of.

I note that all the tests pass if do
```suggestion
          const functionNode = node.value as TSESTree.FunctionExpression;
```

I'm not sure if that's due to missing test coverage or that the function call is actually redundant. Could you investigate? Thanks!

Reply:
I’ve reviewed the code and tested various scenarios, and it seems safe to use type assertion (node.value as TSESTree.FunctionExpression) here. I couldn’t identify any edge cases where findFunctionNode would be necessary. Therefore, I went ahead with the suggested change. Let me know if you have any further concerns! 😊

Reply:
Awesome! Thank you for looking into it! 🙂